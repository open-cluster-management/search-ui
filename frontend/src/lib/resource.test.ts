import {
    getResourceName,
    setResourceName,
    getResourceNamespace,
    setResourceNamespace,
    getResourceApiPath,
    getResourceNameApiPath,
} from './resource'

describe('Resource.ts', () => {
    test('Correctly returns getResourceName', () => {
        const result = getResourceName({
            apiVersion: 'api/v1',
            kind: 'pod',
            metadata: {
                name: 'testPod',
                namespace: 'testNamespace',
            },
        })
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns setResourceName', () => {
        const result = setResourceName(
            {
                apiVersion: 'api/v1',
                kind: 'pod',
                metadata: {
                    name: 'testPod',
                    namespace: 'testNamespace',
                },
            },
            'newName'
        )
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns setResourceName without meta', () => {
        const result = setResourceName(
            {
                apiVersion: 'api/v1',
                kind: 'pod',
            },
            'newName'
        )
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns getResourceNamespace', () => {
        const result = getResourceNamespace({
            apiVersion: 'api/v1',
            kind: 'pod',
            metadata: {
                name: 'testPod',
                namespace: 'testNamespace',
            },
        })
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns setResourceNamespace', () => {
        const result = setResourceNamespace(
            {
                apiVersion: 'api/v1',
                kind: 'pod',
                metadata: {
                    name: 'testPod',
                    namespace: 'testNamespace',
                },
            },
            'newNamespace'
        )
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns setResourceNamespace wihtout meta', () => {
        const result = setResourceNamespace(
            {
                apiVersion: 'api/v1',
                kind: 'pod',
            },
            'newNamespace'
        )
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns getResourceNameApiPath', () => {
        const testData = {
            apiVersion: 'v1',
            kind: 'pod',
            plural: undefined,
            metadata: {
                name: 'testPod',
                namespace: 'testNamespace',
            },
        }
        const result = getResourceNameApiPath(testData)
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns getResourceNameApiPath alt arguments', () => {
        const testData = {
            apiVersion: 'api/v1',
            plural: 'pluralValues',
        }
        const result = getResourceNameApiPath(testData)
        expect(result).toMatchSnapshot()
    })

    test('Correctly returns getResourceNameApiPath no kind', () => {
        const testData = {
            apiVersion: 'v1',
            kind: undefined,
            plural: undefined,
            metadata: {
                name: 'testPod',
                namespace: 'testNamespace',
            },
        }
        const result = getResourceNameApiPath(testData)
        expect(result).toMatchSnapshot()
    })
})
