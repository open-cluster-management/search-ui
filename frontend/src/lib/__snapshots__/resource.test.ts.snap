// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource.ts Correctly returns getResourceName 1`] = `"testPod"`;

exports[`Resource.ts Correctly returns getResourceNameApiPath 1`] = `"/api/v1/namespaces/testNamespace/pods/testPod"`;

exports[`Resource.ts Correctly returns getResourceNameApiPath alt arguments 1`] = `"/apis/api/v1/pluralValues"`;

exports[`Resource.ts Correctly returns getResourceNameApiPath no kind 1`] = `"/api/v1/namespaces/testNamespace/testPod"`;

exports[`Resource.ts Correctly returns getResourceNamespace 1`] = `"testNamespace"`;

exports[`Resource.ts Correctly returns setResourceName 1`] = `"newName"`;

exports[`Resource.ts Correctly returns setResourceName without meta 1`] = `"newName"`;

exports[`Resource.ts Correctly returns setResourceNamespace 1`] = `"newNamespace"`;

exports[`Resource.ts Correctly returns setResourceNamespace wihtout meta 1`] = `"newNamespace"`;
